src/components/FilterBarInline/FilterBarInline.mdx

```diff
@@ -0,0 +1,349 @@
+
+<Source
+  code={`
+const onClearFilterBar = () => {
+  onClearAll();
+  setSelectedCheckbox([]);
+  setTextValue("");
+};`}
+  language="jsx"
+/>
+
+It is expected that the `onSubmit` handler will read the current values from the
+child form elements (the filters) and update the associated search results list
+based on those values.
+
+It is expected that the `onClear` handler will clear all of the current values
+from the child form elements and restore the associated search results list to an
+unfiltered state.
+
+To have better readability of the code it is recommended to separate nested render
+prop functions:
+
+<Source
+  code={`
+// function for renderMultiSelect prop of MultiSelectGroup 
+const renderMultiSelect = ({ isBlockElement, multiSelectWidth }) => {
+  return (
+    shortMultiSelectItems &&
+    shortMultiSelectItems.map((multiSelect) => (
+      <MultiSelect
+        buttonText={multiSelect.name}
+        key={multiSelect.id}
+        id={multiSelect.id}
+        items={multiSelect.items}
+        selectedItems={selectedItems}
+        isBlockElement={isBlockElement}
+        onChange={(e) => onChange(e.target.id, multiSelect.id)}
+        onMixedStateChange={(e) => {
+          return onMixedStateChange(e.target.id, multiSelect.id, multiSelect.items);
+        }}
+        onClear={() => onClear(multiSelect.id)}
+        width={multiSelectWidth}
+      />
+    )) 
+  ); 
+};`}
+  language="jsx"
+/>
+
+<Source
+  code={`
+// function for renderChildren prop of FilterBar 
+const renderFilterComponents = ({ layout, width }) => {
+  return (
+    <>
+      <MultiSelectGroup
+        id="multiselect-group"
+        labelText="MultiSelect Group"
+        layout={layout}
+        multiSelectWidth={width}
+        renderMultiSelect={renderMultiSelect}
+      />
+      <CheckboxGroup
+        id="checkbox-example"
+        labelText="Checkbox Group"
+        name="checkboxExample"
+        layout={layout}
+        value={selectedCheckbox}
+        onChange={(e) => {
+          setSelectedCheckbox(e);
+        }}
+      >
+        <Checkbox id="checkbox-1" value="1" labelText="Checkbox 1" />
+        <Checkbox id="checkbox-2" value="2" labelText="Checkbox 2" />
+      </CheckboxGroup>
+      <TextInput
+        id="textinput-example"
+        isClearable
+        isClearableCallback={() => setTextValue("")}
+        labelText="What is your favorite color?"
+        onChange={(e) => setTextValue(e.target.value)}
+        placeholder="i.e. blue, green, etc."
+        value={textValue}
+      />
+    </>
+  ); 
+};
+`}
+  language="jsx"
+/>
+
+<Source
+  code={`
+return (
+  <FilterBarInline
+    id={args.id}
+    heading={args.heading}
+    layout={args.layout}
+    onClear={onClearFilterBar}
+    onSubmit={() => console.log(selectedFilterItems)}
+    selectedItems={selectedFilterItems}
+    renderChildren={renderFilterComponents}
+  />
+); `}
+  language="jsx"
+/>
+
+## useMultiSelect hook
```

Comment:
I think you can remove this section entirely. You can still mention this and link to the appropriate doc page in the section above near the "To control each of the child form elements...." sentence. At that point, you can say "`useMultiSelect` is an optional but recommended hook ...." and then link to the docs.

It follows the same pattern as "Find documentation on how to do this on each individual component page".